life_expectancy/file_handler.py

```diff
@@ -1,49 +1,138 @@
+
+    def load_data(self, input_file_path: Union[str, Path]) -> pd.DataFrame:
+        """Load the data from a file."""
+        try:
+            with open(Path(input_file_path), encoding="utf-8") as filename:
+                df_raw = pd.read_csv(filename, sep="\t", na_values=[":"])
+        except FileNotFoundError:
+            logging.error("Input file %s not found.", input_file_path)
+            return pd.DataFrame()
+        logging.info("Successfully loaded %s", input_file_path)
+        return df_raw
+
+    def load_data_from_content(self, content: str) -> pd.DataFrame:
+        """Load the data from a string."""
+        return pd.read_csv(StringIO(content), sep="\t", na_values=[":"])
+
+
+class JSONFileLoadingStrategy(FileLoadingStrategy):
+    """Class to load JSON files."""
+
+    def load_data(self, input_file_path: Union[str, Path]) -> pd.DataFrame:
+        """Load the data from a file."""
+        try:
+            with open(Path(input_file_path), encoding="utf-8") as filename:
+                data = json.load(filename)
+                df_raw = pd.DataFrame(data)
+        except FileNotFoundError:
+            logging.error("Input file %s not found.", input_file_path)
+            return pd.DataFrame()
+        logging.info("Successfully loaded %s", input_file_path)
+        return df_raw
+
+    def load_data_from_content(self, content: str) -> pd.DataFrame:
+        """Load the data from a string."""
+        data = json.loads(content)
+        return pd.DataFrame(data)
+
+
+class ZipFileLoadingStrategy(FileLoadingStrategy):
+    """Class to load zip files."""
+
+    def load_data(self, input_file_path: Union[str, Path]) -> pd.DataFrame:
+        try:
+            with zipfile.ZipFile(Path(input_file_path), "r") as zip_ref:
+                for file in zip_ref.namelist():
+                    file_ext = Path(file).suffix.lower()
+                    strategy: Union[
+                        CSVFileLoadingStrategy, JSONFileLoadingStrategy
+                    ] = CSVFileLoadingStrategy()
+                    if file_ext in (".csv", ".tsv"):
+                        strategy = CSVFileLoadingStrategy()
+                    elif file_ext == ".json":
+                        strategy = JSONFileLoadingStrategy()
+                    else:
+                        logging.error("Unsupported file type inside zip: %s", file_ext)
+                        return pd.DataFrame()
+
+                    with zip_ref.open(file, "r") as file_ref:
+                        file_content = file_ref.read().decode("utf-8")
+                        return strategy.load_data_from_content(file_content)
+        except FileNotFoundError:
+            logging.error("Input file %s not found.", input_file_path)
+            return pd.DataFrame()
+        except zipfile.BadZipFile:
+            logging.error("Input file %s is not a valid zip file.", input_file_path)
+            return pd.DataFrame()
+        return pd.DataFrame()
```

Comment:
```suggestion
def load_data(self, input_file_path: Union[str, Path]) -> pd.DataFrame:
    
    if not input_file_path.exists():
        logging.error("Input file %s not found.", input_file_path)
        return pd.DataFrame()
    
    strategies = {
        ".csv": CSVFileLoadingStrategy,
        ".tsv": JSONFileLoadingStrategy
    }
        
    try:
        with zipfile.ZipFile(Path(input_file_path), "r") as zip_ref:
            file = zip_file.namelist()[0]
            file_ext = Path(file).suffix.lower()
            try:
                strategy = strategies[file_ext]()
            except KeyError as err:
                    logging.error("Unsupported file type inside zip: %s", file_ext)
                    raise err
                
            with zip_ref.open(file, "r") as file_ref:
                    file_content = file_ref.read().decode("utf-8")
                    return strategy.load_data_from_content(file_content)

    except zipfile.BadZipFile:
        logging.error("Input file %s is not a valid zip file.", input_file_path)
        return pd.DataFrame()
    
```

I tried to simplify this function:
- Check if the file exists before trying to unzip it
- Use a dictionary instead of an if-else block to remove one indentation level
- Instead of looking all files inside the zip, just open the first one (because the original code was already exiting the loop after the first file read, the results are the same.

Of course, we can always simplify ever further... heheh

For example, instead of 3 strategies: Json, Csv, and Zip, we could use only 2: CSVFileLoadingStrategy and ZippedJsonLoadingStrategy. Your `load_data` could become simply:

```python
def load_data(self, input_file_path: Union[str, Path]) -> pd.DataFrame:
    """Load a Zipped JSON data and returns a pandas DataFrame."""
        with zipfile.ZipFile(input_file_path) as zip_file:
            zipped_fname = zip_file.namelist()[0]
            with zip_file.open(zipped_fname) as json_file:
                df = pd.read_json(json_file)
        return df